### PR TITLE
Fix: Allow for unlimited sample filter using filter by name or id dialog [ENHC0010016]

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Fixes issue caused by using a `get` request to filter the table by list of puids or names.  Get requests are limited to 2,048 characters which prevented this.  Changed the form to use a `post` request.  To facilitate this, I needed to:

1. Update the search form to surround the entire fields and table.  This was required because the pagination requests needed to use the data in the search fields and therefore must also be submitted as a `post`
2. Updated the select all to make a copy of the search form and update the URL for search, and then request the submission of this new form.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

No UI changes were required.

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Go to any project that has samples
2. Try "Filter by sample name or id", I created a text file with 300 random sample names and a couple from the project I was in to ensure that the filter was working and could handle a large amount of text.
3. Try pagination with the filter
4. Ensure selection and select all is working

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by ChrisHuynh333 — APPROVED** [review]
Works great! Nothing to add on my end

File: app/components/list_filter_component.rb [hunk 1]
```diff
@@ -2,9 +2,10 @@
 
 # View component to add a dialog for filter table by a list
 class ListFilterComponent < Component
-  attr_reader :form
+  attr_reader :form, :samples
```
Comment (ericenns) [inline app/components/list_filter_component.rb:5]: **nitpick (blocking):** Can you rename `samples` to `filters`, as it really isn't a list of samples but a list of filters.

File: app/components/ransack/sort_post_component.rb
Comment (ericenns) [inline app/components/ransack/sort_post_component.rb:1]: **praise (blocking):** Is this component required? Since we are no longer using post requests for sort I believe this component and associated files can be removed.

File: app/controllers/concerns/sample_actions.rb
Comment (ericenns) [inline app/controllers/concerns/sample_actions.rb:1]: **nitpick (blocking):** Can you revert the changes to this file

File: app/controllers/groups/samples_controller.rb [hunk 2]
```diff
@@ -63,21 +62,32 @@ def current_page
...
+    end
+
+    def process_samples
+      authorize! @group, to: :sample_listing?
+      @search_params = search_params
+      set_metadata_fields
 
-      @q.sorts = 'updated_at desc' if @q.sorts.empty?
+      @q = authorized_samples.ransack(@search_params)
+      @pagy, @samples = pagy_with_metadata_sort(@q.result)
```
Comment (ericenns) [inline app/controllers/groups/samples_controller.rb:75]: **nitpick (blocking):** Can you move the last 2 lines of this method into the index as it is not required for `search`

File: app/controllers/projects/samples_controller.rb [hunk 3]
```diff
@@ -123,24 +122,36 @@ def current_page
...
-    def set_search_params
-      @search_params = params[:q].nil? ? {} : params[:q].to_unsafe_h
+    def process_samples
+      authorize! @project, to: :sample_listing?
+
+      @search_params = search_params
+
+      set_metadata_fields
+      @q = load_samples.ransack(@search_params)
+      @pagy, @samples = pagy_with_metadata_sort(@q.result)
```
Comment (ericenns) [inline app/controllers/projects/samples_controller.rb:143]: **nitpick (blocking):** Can you move the last 2 lines of this method into the index as it is not required for `search`

File: app/helpers/sorting_helper.rb [hunk 4]
```diff
@@ -17,12 +17,17 @@ def sorting_item(dropdown, ransack_obj, field, dir)
                        })
   end
 
-  def sorting_url(ransack_obj, field, dir: nil)
-    url = if dir.nil?
-            sort_url(ransack_obj,
-                     field).to_s
+  def sorting_url(ransack_obj, field, dir: nil, with_params: true)
```
Comment (ericenns) [inline app/helpers/sorting_helper.rb:20]: **nitpick (blocking):** Can you rename `with_params` to `with_search_params`.

File: config/initializers/session_store.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+
+Rails.application.config.session_store :active_record_store, key: '_iria_next_session'
```
Comment (ericenns) [inline config/initializers/session_store.rb:3]: **nitpick (blocking):** can you change this to `_irida_next_session`

**Review by ericenns — CHANGES_REQUESTED** [review]
Working as expected, just a few comments

File: app/components/list_filter_component.rb (hunk 1 above)
Comment (joshsadam) [reply app/components/list_filter_component.rb:5]: Updated in 42bfa4ec7dfbae6991a2175289794d8c0d0d4f03

File: app/components/ransack/sort_post_component.rb
Comment (joshsadam) [reply app/components/ransack/sort_post_component.rb:1]: Removed in b31f1272c3ce8c2c1a5e9771459d3432c50946e7

File: app/controllers/concerns/sample_actions.rb
Comment (joshsadam) [reply app/controllers/concerns/sample_actions.rb:1]: Reverted in 8ad1c0588b13a62f68c873c909f202cdade25bb1

File: app/controllers/groups/samples_controller.rb (hunk 2 above)
Comment (joshsadam) [reply app/controllers/groups/samples_controller.rb:75]: Moved in 8ffdb7678b9b118ee9c4705612df765ea501a048

File: app/controllers/projects/samples_controller.rb (hunk 3 above)
Comment (joshsadam) [reply app/controllers/projects/samples_controller.rb:143]: Moved in 8ffdb7678b9b118ee9c4705612df765ea501a048

File: app/helpers/sorting_helper.rb (hunk 4 above)
Comment (joshsadam) [reply app/helpers/sorting_helper.rb:20]: Updated in 1e667007069ab6bb597cd8aabd077aad808cdbde

File: config/initializers/session_store.rb (hunk 5 above)
Comment (joshsadam) [reply config/initializers/session_store.rb]: Fixed in b39da32

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me and works great!
